### PR TITLE
Update packages (+ migration)

### PR DESCRIPTION
**Pull request by jicata**

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/RolesController.cs [hunk 1]
```diff
@@ -3,11 +3,12 @@
     using System;
     using System.Collections;
     using System.Linq;
+    using System.Web;
     using System.Web.Mvc;
-
+    using Data.Models;
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/RolesController.cs]: Split using statements.

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/RolesController.cs [hunk 2]
```diff
@@ -3,11 +3,12 @@
...
     using Kendo.Mvc.Extensions;
     using Kendo.Mvc.UI;
-
+    using Microsoft.AspNet.Identity;
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/RolesController.cs]: Split using statements.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 3]
```diff
@@ -4,18 +4,25 @@
     using System.Globalization;
     using System.Linq;
     using System.Web.Mvc;
+    using Microsoft.AspNet.Identity;
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: Split using statements.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 4]
```diff
@@ -69,11 +77,12 @@ public ContentResult GetPointsByEmail(string apiKey, int? contestId, string emai
                 return this.Content("ERROR: Invalid arguments");
             }
 
+            var userIsAdmin = this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
             var isValidApiKey = this.data.Users
                 .All()
                 .Any(x => x.Id == apiKey &&
-                    (x.Roles.Any(y => y.Role.Name == GlobalConstants.AdministratorRoleName) ||
-                    x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
+                          (userIsAdmin ||
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: One tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 5]
```diff
@@ -69,11 +77,12 @@ public ContentResult GetPointsByEmail(string apiKey, int? contestId, string emai
...
+                           x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: One tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 6]
```diff
@@ -111,11 +120,12 @@ public JsonResult GetAllResultsForContest(string apiKey, int? contestId)
                 return this.Json(new ErrorMessageViewModel("Invalid arguments"), JsonRequestBehavior.AllowGet);
             }
 
+            var userIsAdmin = this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
             var isValidApiKey = this.data.Users
                 .All()
                 .Any(x => x.Id == apiKey &&
-                    (x.Roles.Any(y => y.Role.Name == GlobalConstants.AdministratorRoleName) ||
-                    x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
+                          (userIsAdmin ||
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: One tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 7]
```diff
@@ -111,11 +120,12 @@ public JsonResult GetAllResultsForContest(string apiKey, int? contestId)
...
+                           x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: One tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 8]
```diff
@@ -158,11 +168,12 @@ public JsonResult GetAllUserResultsPercentageByForContest(string apiKey, int? co
                 return this.Json(new ErrorMessageViewModel("Invalid arguments"), JsonRequestBehavior.AllowGet);
             }
 
+            var userIsAdmin = this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
             var isValidApiKey = this.data.Users
                 .All()
                 .Any(x => x.Id == apiKey &&
-                    (x.Roles.Any(y => y.Role.Name == GlobalConstants.AdministratorRoleName) ||
-                    x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
+                          (userIsAdmin ||
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: One tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 9]
```diff
@@ -158,11 +168,12 @@ public JsonResult GetAllUserResultsPercentageByForContest(string apiKey, int? co
...
+                           x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: One tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 10]
```diff
@@ -203,11 +214,12 @@ public JsonResult GetAllResultsForContestWithPoints(string apiKey, int? contestI
                 return this.Json(new ErrorMessageViewModel("Invalid arguments"), JsonRequestBehavior.AllowGet);
             }
 
+            var userIsAdmin = this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
             var isValidApiKey = this.data.Users
                 .All()
                 .Any(x => x.Id == apiKey &&
-                    (x.Roles.Any(y => y.Role.Name == GlobalConstants.AdministratorRoleName) ||
-                    x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
+                          (userIsAdmin ||
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: One tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 11]
```diff
@@ -203,11 +214,12 @@ public JsonResult GetAllResultsForContestWithPoints(string apiKey, int? contestI
...
+                           x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: One tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/SubmissionsController.cs [hunk 12]
```diff
@@ -2,10 +2,14 @@
 {
     using System.Linq;
     using System.Web.Mvc;
+    using Microsoft.AspNet.Identity;
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/SubmissionsController.cs]: Split using statements.

**Comment by KristianMariyanov** [comment]
The generalization of @Innos CR is: @jicata just setup your resharper man 😆

**Comment by jicata** [comment]
More specific guidelines are welcome :)

**Comment by KristianMariyanov** [comment]
I will help you with that next time when I see you :)

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 13]
```diff
@@ -26,11 +34,12 @@ public ContentResult GetPointsByAnswer(string apiKey, int? contestId, string ans
                 return this.Content("ERROR: Invalid arguments");
             }
 
+            var userIsAdmin = this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
             var isValidApiKey = this.data.Users
                 .All()
                 .Any(x => x.Id == apiKey &&
-                    (x.Roles.Any(y => y.Role.Name == GlobalConstants.AdministratorRoleName) ||
-                    x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
+                        (userIsAdmin ||
```
Comment (YanaSlavcheva) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: shift + tab

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 14]
```diff
@@ -69,11 +78,12 @@ public ContentResult GetPointsByEmail(string apiKey, int? contestId, string emai
...
```
Comment (YanaSlavcheva) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: shift + tab

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 15]
```diff
@@ -111,11 +121,12 @@ public JsonResult GetAllResultsForContest(string apiKey, int? contestId)
                 return this.Json(new ErrorMessageViewModel("Invalid arguments"), JsonRequestBehavior.AllowGet);
             }
 
+            var userIsAdmin = this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
             var isValidApiKey = this.data.Users
                 .All()
                 .Any(x => x.Id == apiKey &&
-                    (x.Roles.Any(y => y.Role.Name == GlobalConstants.AdministratorRoleName) ||
-                    x.LecturerInContests.Any(y => y.ContestId == contestId.Value)));
+                        (userIsAdmin ||
```
Comment (YanaSlavcheva) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: shift + tab

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 16]
```diff
@@ -158,11 +169,12 @@ public JsonResult GetAllUserResultsPercentageByForContest(string apiKey, int? co
...
```
Comment (YanaSlavcheva) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: shift + tab

File: Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs [hunk 17]
```diff
@@ -203,11 +215,12 @@ public JsonResult GetAllResultsForContestWithPoints(string apiKey, int? contestI
...
```
Comment (YanaSlavcheva) [inline Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs]: shift + tab

File: Open Judge System/Data/OJS.Data/Migrations/201708220830272_identity_update.cs [hunk 18]
```diff
@@ -0,0 +1,56 @@
+namespace OJS.Data.Migrations
+{
+    using System;
+    using System.Data.Entity.Migrations;
+    
+    public partial class identity_update : DbMigration
+    {
+        public override void Up()
+        {
+            RenameColumn(table: "dbo.AspNetUserClaims", name: "User_Id", newName: "UserId");
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/Migrations/201708220830272_identity_update.cs]: All these rows must have `this.` in their beginning. We really must set up ur ReSharper... :)

File: Open Judge System/Data/OJS.Data/Migrations/201708220830272_identity_update.cs [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+            this.AlterColumn("dbo.AspNetRoles", "Name", c => c.String(nullable: false, maxLength: 256));
+            this.AddPrimaryKey("dbo.AspNetUserLogins", new[] { "LoginProvider", "ProviderKey", "UserId" });
+            this.CreateIndex("dbo.AspNetUsers", "UserName", unique: true, name: "UserNameIndex");
+            this.CreateIndex("dbo.AspNetRoles", "Name", unique: true, name: "RoleNameIndex");
+            this.DropColumn("dbo.AspNetUsers", "Discriminator");
+        }
+        
+        public override void Down()
+        {
+            AddColumn("dbo.AspNetUsers", "Discriminator", c => c.String(nullable: false, maxLength: 128));
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/Migrations/201708220830272_identity_update.cs]: `this.`

File: Open Judge System/Data/OJS.Data/Migrations/201708220830272_identity_update.cs [hunk 20]
```diff
@@ -0,0 +1,56 @@
+namespace OJS.Data.Migrations
+{
+    using System;
+    using System.Data.Entity.Migrations;
+
+    public partial class identity_update : DbMigration
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/Migrations/201708220830272_identity_update.cs]: for future reference - you might want to use `PascalCase` when naming Migrations - this is a `.cs` file in its essence. CS files we name in PascalCase.

**Pull request merged**